django/core/paginator.py

```diff
@@ -54,6 +57,16 @@ def page(self, number):
             top = self.count
         return self._get_page(self.object_list[bottom:top], number, self)
 
+    def page_querystring(self, page_number):
+        """
+        Returns a query string for the given page number, preserving any
+        GET parameters present.
+        """
+        params = self.request.GET.copy() \
+            if self.request else QueryDict(mutable=True)
```

Comment:
why is `mutable=True` needed? No tests fail if I remove it -- we should have a test for it if it's required.